mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1070,19 +1075,19 @@ impl TakerSwap {
         };
 
         // This will be done during order match
-        self.w().watcher_reward = false;
+        self.w().watcher_reward = std::env::var("USE_WATCHER_REWARD").is_ok();
```

Comment:
We should avoid reading a system variable like this throughout the execution of the program. Ideally it should be read only once while initializing mm2 or possibly while enabling specific coins.